Library/Homebrew/os.rb

```diff
@@ -13,6 +13,19 @@ def self.linux?
     RbConfig::CONFIG["host_os"].include? "linux"
   end
 
+  def self.family
+    :generic
+  end
+
+  class << self
+    def build_system_info
+      {
+        "os" => OS.family,
```

Comment:
Could use `HOMEBREW_SYSTEM` or `RbConfig::CONFIG["host_os"]` for this maybe?

Reply:
`RbConfig::CONFIG["host_os"]` returns the Darwin version so that would require modification to work.

`ENV["HOMEBREW_SYSTEM"]` is possible. It would return `Macintosh` and `Linux`. We could maybe also use `ENV["HOMEBREW_OS_VERSION"]` for `os_version` rather than the Ruby stuff - depending on what format is preferred.

Reply:
> `ENV["HOMEBREW_SYSTEM"]` is possible. It would return `Macintosh` and `Linux`. We could maybe also use `ENV["HOMEBREW_OS_VERSION"]` for `os_version` rather than the Ruby stuff - depending on what format is preferred.

Yeh, either is fine. Just thinking of how to keep more logic out of `os.rb`.